core/src/program/blocks/span_block.rs

```diff
@@ -52,7 +54,27 @@ impl Span {
     pub fn new(operations: Vec<Operation>) -> Self {
         assert!(!operations.is_empty()); // TODO: return error
         let (op_batches, hash) = batch_ops(operations);
-        Self { op_batches, hash }
+        Self {
+            op_batches,
+            hash,
+            decorators: DecoratorList::new(),
+        }
+    }
+
+    /// Returns a new [Span] block instantiated with the specified operations and decorators.
+    ///
+    /// # Errors (TODO)
+    /// Returns an error if:
+    /// - `operations` vector is empty.
+    /// - `operations` vector contains any number of system operations.
+    pub fn with_decorators(operations: Vec<Operation>, decorators: DecoratorList) -> Self {
+        assert!(!operations.is_empty()); // TODO: return error
```

Comment:
Would be good to add checks (in debug mode only) to make sure the decorator list is valid (i.e., sorted in ascending order, the last index is not greater than `operations.len()`).

Reply:
Debug flag is not a part of assembler yet. It was meant to be added with debug related decorators like AsmOp. Should the debug flag be added in this PR or the next one with the introduction of AsmOp decorator and we can add validity checks after that flag is added?

Reply:
I meant Rust's debug mode - e.g., via `#[cfg(debug_assertions)]`.

Reply:
Oh ok.

Reply:
Done